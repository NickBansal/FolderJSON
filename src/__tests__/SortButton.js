import "../setupTest";
import React from "react";
import SortButton from "../Components/SortButtons";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe.only("<SortButton />", () => {
  let wrapper;
  const sortFolders = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<SortButton sortFolders={sortFolders} />);
  });

  it("Component matches the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should call the onClick function with name", () => {
    const nameButton = wrapper.find("button");
    nameButton.first().simulate("click");
    expect(sortFolders).toBeCalledWith("name");
  });

  it("Should call the onClick function with added", () => {
    const nameButton = wrapper.find("button");
    nameButton.at(1).simulate("click");
    expect(sortFolders).toBeCalledWith("added");
  });

  it("Should call the onClick function with size", () => {
    const nameButton = wrapper.find("button");
    nameButton.last().simulate("click");
    expect(sortFolders).toBeCalledWith("size");
  });
});
