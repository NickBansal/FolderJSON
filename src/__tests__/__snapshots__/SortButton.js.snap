// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortButton /> Component matches the snapshot 1`] = `
<div
  className="Buttons"
>
  <div
    key="0"
  >
    <button
      className="Buttons__individual"
      onClick={[Function]}
    >
      name
    </button>
  </div>
  <div
    key="1"
  >
    <button
      className="Buttons__individual"
      onClick={[Function]}
    >
      added
    </button>
  </div>
  <div
    key="2"
  >
    <button
      className="Buttons__individual"
      onClick={[Function]}
    >
      size
    </button>
  </div>
</div>
`;
